### PR TITLE
mask operand according to its size

### LINKED ISSUE #110: shellcode hashes operand size issue
For
```asm
mov     r8d, 95902B19h
```
IDAPython returns (MSB is 1)
```
Python>get_operand_value(here(), 1)
0xffffffff95902b19
```

This values is then queried in the database and not found resulting in no annotation.

For `mov     r8d, 8FE060Ch` IDAPython returns 0x8fe060c (MSB is 0) and everything works as expected.

PR suggestion for this incoming.

### PR DESCRIPTION
**Pull request by mr-tz**
closes #110

not updating legacy IDA API, but happy to do so, if you insist 😉

### CODE REVIEW

File: python/flare/shellcode_hash_search.py [hunk 1]
```diff
@@ -342,6 +351,9 @@ def lookForOpArgs(self, start, end):
                     if t == idc.o_imm:
                         if using_ida7api:
                             opval = idc.get_operand_value(head, i)
+                            insn = idautils.DecodeInstruction(head)
+                            opmask = OPERAND_MASK[insn.ops[i].dtype]
```
Comment (jhsmith) [inline python/flare/shellcode_hash_search.py:355]: There are other possible dt_type values that we don't care about (dt_float, dt_double). Change to OPERAND_MASK.get() & check for None, in which case we skip masking.

**Review by jhsmith — CHANGES_REQUESTED** [review]
Just a minor change - otherwise looks great

File: python/flare/shellcode_hash_search.py (hunk 1 above)
Comment (mr-tz) [reply python/flare/shellcode_hash_search.py]: addressed in 083f743, thanks!

**Review by jhsmith — APPROVED** [review]
Awesome!

**Pull request merged**